addon/test-waiter.ts

```diff
@@ -0,0 +1,116 @@
+import { IWaiter, WaiterName, ITestWaiterDebugInfo } from './types';
+import { register } from './waiter-manager';
+
+/**
+ * A class providing creation, registration and async waiting functionality.
+ *
+ * @public
+ * @class SimpleWaiter<T>
+ *
+ * @example
+ *
+ * import Component from '@ember/component';
```

Comment:
Specifically, these leading asterisks make the examples hard to review and break the ability to copy paste.